### DISCUSSION TITLE
Image capture very slow to succeed

### DISCUSSION THREAD

**Discussion by ravenfeld** [ujizin/Camposer#52] [Q&A]
Hello, I find that it takes a long time to receive the success callback.
I read a lot of issues
https://groups.google.com/a/android.com/g/camerax-developers/c/PTeOQHxIK7c
https://groups.google.com/a/android.com/g/camerax-developers/c/d96KKpNfuGg/m/oKZeiZQsBgAJ

Have you noticed this too?

**Comment by ujizin** [comment]
Hey @ravenfeld, yes, I’ve noticed that as well. By the way, CameraX recently released a feature to minimize this latency, and it can be used in Camposer too. If you want to try it out, please use ImplementationMode.ZERO_SHUTTER_LAG in the CameraPreview composable.

It still has some delay, but the goal is to reduce it a bit. I hope it meets your expectations.

Thanks for starting this discussion here! 😄

**Reply by ravenfeld** [reply]
I've also tested it, but as you can see, it's not as responsive as my camera's native photo application.

**Reply by ravenfeld** [reply]
For your information, it works better without flash.
